### PR TITLE
Localization model added to DateExtension 

### PR DESCRIPTION
**Pull request by isshort**
Hello.
This is the solution to https://github.com/VB10/kartal/pull/49 can you please check

### CODE REVIEW

File: lib/src/date_extension.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    return '';
+  }
+}
+
+/// The code snippet defines an extension method `timeDiff` on the `DateTime` class. This extension
+/// method returns a custom extension object `_DateExtension` that calculates the time difference
+/// between the current time and a given time.
+
+extension DateTimeExtension1 on DateTime {
```
Comment (VB10) [inline lib/src/date_extension.dart:37]: the name is not looking good. please update this

File: lib/src/date_extension.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  /// represents the current time. If a value is provided, it will be used as the current time for
+  /// calculating the time difference. If no value is provided, the current system time will be used.
+  _DateExtension timeDiff({
+    DateTime? currentTime,
+    DateDifferenceModel? dateDifferenceModel,
+  }) =>
+      _DateExtension(
+        currentTime: currentTime ?? DateTime.now(),
+        dateTime: this,
+        dateDifferenceModel: dateDifferenceModel ?? const DateDifferenceModel(),
```
Comment (VB10) [inline lib/src/date_extension.dart:52]: verıable name is not good

currentTime and DateTime is mean like a now.

so u can change the name like a  other time or difrreance Time

File: lib/src/date_extension.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  ///   currentTime (DateTime): The `currentTime` parameter is a nullable `DateTime` object that
+  /// represents the current time. If a value is provided, it will be used as the current time for
+  /// calculating the time difference. If no value is provided, the current system time will be used.
+  _DateExtension timeDiff({
+    DateTime? currentTime,
+    DateDifferenceModel? dateDifferenceModel,
+  }) =>
+      _DateExtension(
+        currentTime: currentTime ?? DateTime.now(),
+        dateTime: this,
```
Comment (VB10) [inline lib/src/date_extension.dart:51]: it name will be currentTime becuase u write a extension on current

File: lib/src/date_extension.dart [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+/// between the current time and a given time.
+
+extension DateTimeExtension1 on DateTime {
+  /// The function returns a custom extension object that calculates the time difference between the
+  /// current time and a given time.
+  ///
+  /// Args:
+  ///   currentTime (DateTime): The `currentTime` parameter is a nullable `DateTime` object that
+  /// represents the current time. If a value is provided, it will be used as the current time for
+  /// calculating the time difference. If no value is provided, the current system time will be used.
```
Comment (VB10) [inline lib/src/date_extension.dart:44]: documentation is not clear, please use more understandable

File: lib/src/date_extension.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  _DateExtension timeDiff({
+    DateTime? currentTime,
```
Comment (VB10) [inline lib/src/date_extension.dart:46]: current time have to be here so please change to nullable and set to defaullt value lika a now

File: lib/src/date_extension.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    final minuteDifference = _currentTime.difference(_dateTime).inMinutes;
+    if (dayDifference > 0) {
+      return '$dayDifference ${_dateDifferenceModel.dayDifferenceLabel}}';
+    } else if (hourDifference > 0) {
+      return '$hourDifference ${_dateDifferenceModel.hourDifferenceLabel}';
+    } else if (minuteDifference > 0) {
+      return '$minuteDifference ${_dateDifferenceModel.minuteDifferenceLabel}';
+    }
+    return '';
+  }
```
Comment (VB10) [inline lib/src/date_extension.dart:30]: change to business beucase it not accaptable. For example dayDifrrence is bigger than zero, your code trying to parse every diff for each item. This need to calculate after the check business

final dayDifference = _currentTime.difference(_dateTime).inDays;
example
if(day > 0) return '';

File: lib/src/date_extension.dart [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    final hourDifference = _currentTime.difference(_dateTime).inHours;
+    final minuteDifference = _currentTime.difference(_dateTime).inMinutes;
+    if (dayDifference > 0) {
+      return '$dayDifference ${_dateDifferenceModel.dayDifferenceLabel}}';
+    } else if (hourDifference > 0) {
+      return '$hourDifference ${_dateDifferenceModel.hourDifferenceLabel}';
+    } else if (minuteDifference > 0) {
+      return '$minuteDifference ${_dateDifferenceModel.minuteDifferenceLabel}';
+    }
+    return '';
```
Comment (VB10) [inline lib/src/date_extension.dart:29]: if the diffrence is seconds, it can not show. Please check this

**Review by VB10 — CHANGES_REQUESTED** [review]
Need to update, end please look at other extension class because i'm using ''.ext in new versions.

**Comment by isshort** [comment]
#60

**Pull request closed without merging**